pkg/resources/scim_integration_acceptance_test.go

```diff
@@ -1,79 +1,180 @@
 package resources_test
 
 import (
-	"fmt"
 	"testing"
 
 	acc "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/snowflakeroles"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
 
+	"github.com/hashicorp/terraform-plugin-testing/config"
 	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
 	"github.com/hashicorp/terraform-plugin-testing/tfversion"
 )
 
-func TestAcc_ScimIntegration(t *testing.T) {
-	scimIntName := acc.TestClient().Ids.Alpha()
-	scimProvisionerRole := "AAD_PROVISIONER"
-	scimNetworkPolicy := acc.TestClient().Ids.Alpha()
+func TestAcc_ScimIntegration_basic(t *testing.T) {
+	networkPolicy, networkPolicyCleanup := acc.TestClient().NetworkPolicy.CreateNetworkPolicy(t)
+	t.Cleanup(networkPolicyCleanup)
 
+	role, roleCleanup := acc.TestClient().Role.CreateRoleWithRequest(t, sdk.NewCreateRoleRequest(snowflakeroles.GenericScimProvisioner).WithOrReplace(true))
+	t.Cleanup(roleCleanup)
+	acc.TestClient().Role.GrantRoleToCurrentRole(t, role.ID())
+	role2, role2Cleanup := acc.TestClient().Role.CreateRoleWithRequest(t, sdk.NewCreateRoleRequest(snowflakeroles.OktaProvisioner).WithOrReplace(true))
+	t.Cleanup(role2Cleanup)
+	acc.TestClient().Role.GrantRoleToCurrentRole(t, role2.ID())
+
+	id := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	m := func(enabled bool, scimClient sdk.ScimSecurityIntegrationScimClientOption, runAsRole *sdk.Role, complete bool) map[string]config.Variable {
+		c := map[string]config.Variable{
+			"name":        config.StringVariable(id.Name()),
+			"enabled":     config.BoolVariable(enabled),
+			"scim_client": config.StringVariable(string(scimClient)),
+			"run_as_role": config.StringVariable(runAsRole.ID().Name()),
+		}
+		if complete {
+			c["sync_password"] = config.BoolVariable(false)
+			c["network_policy_name"] = config.StringVariable(networkPolicy.Name)
+			c["comment"] = config.StringVariable("foo")
+		}
+		return c
+	}
 	resource.Test(t, resource.TestCase{
 		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
 		PreCheck:                 func() { acc.TestAccPreCheck(t) },
 		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
 			tfversion.RequireAbove(tfversion.Version1_5_0),
 		},
-		CheckDestroy: nil,
 		Steps: []resource.TestStep{
 			{
-				Config: scimIntegrationConfigAzure(scimIntName, scimProvisionerRole, scimNetworkPolicy),
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_ScimIntegration/basic"),
+				ConfigVariables: m(false, sdk.ScimSecurityIntegrationScimClientGeneric, role, false),
+				Check: resource.ComposeTestCheckFunc(
```

Comment:
let's check the computed fields too

Reply:
The only check that was missing was TestCheckResourceAttrSet with run_as_role - added.